src/actions/advancement.cpp

```diff
@@ -186,26 +220,30 @@ namespace
 			{
 				res = randomness::generator->get_random_int(0, nb_options_-1);
 
+				// if we are called from an [event] (for example [unstoire_unit] triggered an
+				// advancement of an ai unit during an ai turn) then ai_advancement_ will be empty.
+				// TODO: can we now remove the ai_advancement_ paraemter from this function or are there cases
+				// where this retuns something different than  ai_advancement_ ?
```

Comment:
Please fix the types - "unstore_unit", "parameter", "returns" (the first one is especially important).

Also I think you probably can remove the `ai_advancement_` parameter now (surely it was loaded using the same method you used here?).